### DISCUSSION TITLE
How to use the Touch_RA8875.cpp library

### DISCUSSION THREAD

**Discussion by gwiethaus** [lovyan03/LovyanGFX#631] [Device Configurations]
Hello everyone, 

[tobi807](https://github.com/tobi807) implements a touch screen (Touch_RA8875.cpp) for the RA8875, which is the native resistive screen of the RA8875 integrated circuit.

This implementation is in: (https://github.com/tobi807/LovyanGFX/tree/1b029ef324f3ee53e3f724879ec07a302d402249)

I would like to know how to configure it? I believe that you only need the INT pin, since MISO, MOSI, CLK and CS of the SPI must be the same as the display, that is, the same communication path. I use ESP32-S3 N16R8

**Comment by tobozo** [comment]
Hi,

BlyngGO has a [config example](https://github.com/BlynkGO/BlynkGOv3/blob/main/src/utils/BlynkGO_LCD/BlynkGO_LCD.h) (with touch pins definition [here](https://github.com/BlynkGO/BlynkGOv3/blob/main/src/config/blynkgo_system_config.h#L1770-L1775)), only it's for multiple displays so you'll have to cherry-pick :)

You can look for the `RA8875_DRIVER` and `TOUCH_GSL1680F` macro blocks in both files to identify what to grab.

**Reply by gwiethaus** [reply]
Hello @tobozo,

Sorry for my English.

Thanks for answering. I've been looking at what you sent me and it's still not very clear. The issue is that the RA8875 already has an internal controller for the resistive touch screen. That's why it doesn't have any external controller. I believe it needs the INT pin of the display, because that's the only way to know that a touch occurred on the screen.

I'm using the ESP32-S3 and I set GPIO18 as the interrupt.

I'm using that LovyanGFX driver (implementation by @tobi807) file touch_RA8875.h.

I didn't have much success, because I have to pass several parameters using _touch_instance.config(touch_cfg);

I tried to use the same SPI as the touch, as there is no specific SPI for the resistive TFT, as the one that controls the screen is the RA8875.

I'm not sure if I need to use this block, for example:

```

...
#define TOUCH_MISO 13                      //GPIO 13
#define TOUCH_CLK  12                      //GPIO 12
#define TOUCH_MOSI 11                      //GPIO 11
#define TOUCH_CS  10                      //GPIO 10
#define TOUCH_INT  18                      //GPIO 18
...

class ....
....
lgfx::Touch_RA8875 _touch_instance;
...
{
auto touch_cfg = _touch_instance.config();
touch_cfg.x_min = 0;
touch_cfg.x_max = TFT_WIDTH-1;
touch_cfg.y_min = 0;
touch_cfg.y_max = TFT_HEIGHT-1;
touch_cfg.pin_int = TOUCH_INT;
touch_cfg.bus_shared = true;
touch_cfg.offset_rotation = 0;

//I don't know if this part is necessary, since there is no SPI for the TFT. If I don't put this, it generates an exception

touch_cfg.spi_host = SPI2_HOST;
touch_cfg.freq = 1000000; // SPI
touch_cfg.pin_sclk = TOUCH_SCLK; // SCLK
touch_cfg.pin_mosi = TOUCH_MOSI; // MOSI
touch_cfg.pin_miso = TOUCH_MISO; // MISO
touch_cfg.pin_cs = TOUCH_CS;
....
_touch_instance.config(touch_cfg);
_panel_instance.setTouch(&_touch_instance);
}
....

```

I tried to look for a solution on the internet and in forums and I didn't find anything about it. I used a code from sumotoy/ra8875 and the touch screen worked correctly. Only with lovyanGFX it's not working. The advantage of lovyanGFX is that it's a very fast library for my display to render the screens in the flush callback event, but sumotoy's is terrible.

I would really like to solve this problem and continue using lovyanGFX. If you can help me with this, I would appreciate your help in advance.

**Comment by gwiethaus** [comment]
Sorry for my English.

I'm redoing the Touch_RA8875 library for resistive touch screens. The RA8875 controller itself controls this screen without requiring any other circuit. I believe this was the idea when it was originally created by @tobi807. I analyzed the library and found that it cannot use another SPI channel, since the display does not have one. It makes no sense to connect to a specific SPI of the MCU, in this case the ESP32. Only the INT pin of the RA8875 should be used to receive the occurrence of events.

I analyzed the code and found problems. The internal register of the RA8875 is not sending the value of the x,y position of the screen, but it is triggering the interrupt event and indicating that there was a touch. I'm checking this and rewriting some of the code. I'll leave it according to the LovyanGFX development standard.

With LovyanGFX, sending data to the display was very fast. Despite this, the implementation of the touch screen never had or always failed. I saw @tobi807's code that started the project, but I don't understand why it is not in the touch folder of the LovyanGFX project.

In addition, I am also basing myself on another native project that is on buydisplay and sumotoy among others to help in this journey.

I hope that @tobozo, @okuoku, @lefty01, @zingo, among others, can help me in this project to adjust the RA8875 touch screen.
